src/hooks/use-stable-location.ts

```
@@ -1,20 +1,20 @@
-		}
-		return matchingLocation.current;
-	}, [isLoading, pathname]);
+  const { isLoading, pathname } = useRouterState({
+    select: (state) => ({
+      isLoading: state.isLoading,
+      pathname: state.location.pathname,
+    }),
+  })
+  const matchingLocation = useRef<string>(pathname)
+  return useMemo(() => {
+    if (!isLoading && matchingLocation.current !== pathname) {
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/hooks/use-stable-location.ts:13:23
  11 |   const matchingLocation = useRef<string>(pathname)
  12 |   return useMemo(() => {
> 13 |     if (!isLoading && matchingLocation.current !== pathname) {
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^ Cannot access ref value during render
  14 |       matchingLocation.current = pathname
  15 |     }
  16 |     return matchingLocation.current

(react-hooks/refs)